APIs/openEO/authentication.qmd

```diff
@@ -0,0 +1,198 @@
+title: "Authentication in openEO"
+---
+
+While basic discovery of openEO collections and processes is possible without authentication, executing openEO workflows requires user authentication. This is necessary to manage user quotas, resources, and credit consumption effectively. User authentication in openEO is handled with the OpenID Connect protocol (often abbreviated as "OIDC").
+
+The openEO endpoint of the Copernicus Data Space Ecosystem is configured to use the identity provider service of the Copernicus Data Space Ecosystem. It is therefore recommended to complete the [Copernicus Data Space Ecosystem registration](../../Registration.qmd) before attempting to authenticate with openEO.
+
+
+
+
+## Getting Started: A Typical openEO Authentication Flow With The openEO Python Client Library {#sec-typical-authentication-flow}
+
+A typical Copernicus Data Space Ecosystem openEO workflow,
+using the openEO Python client library,
+starts with setting up a connection like this:
+
+```python
+import openeo
+
+connection = openeo.connect(url="openeo.dataspace.copernicus.eu")
+connection.authenticate_oidc()
+```
+
+After connecting to the Copernicus Data Space Ecosystem openEO endpoint,
+[`Connection.authenticate_oidc()`](https://open-eo.github.io/openeo-python-client/api.html#openeo.rest.connection.Connection.authenticate_oidc){target="_blank"} initiates the OpenID Connect authentication flow:
+
+-   By default, the first time `authenticate_oidc()` is called,
+    instructions to visit a certain URL will be printed, e.g.:
+
+    ```
+    Visit https://auth.example/?user_code=EAXD-RQXV to authenticate.
+    ```
+
+    Visit this URL (click or copy-paste it into a web browser) and follow the login flow using Copernicus Data Space Ecosystem credentials.
+
+    ::: {.callout-tip collapse="false"}
+
+    Visit this URL using any preferred browser to complete the login procedure (e.g., on a laptop or smartphone). It does *not* need to be a browser on the same machine or network as the Python script/application.
+
+    :::
+
+    Once authentication is complete, the Python script will receive the necessary authentication tokens and print
+
+    ```
+    Authorized successfully.
+    ```
+
+-   Other times, when valid (refresh) tokens are still present on the system, the openEO Python client library will automatically use these tokens and bypass the manual login process. The following will be visible immediately:
+
+    ```
+    Authenticated using refresh token.
+    ```
+
+In any case, `connection` is now authenticated and ready to make download or processing requests.
+
+
+## Alternative Authentication Methods
+
+The [openEO Python client library documentation](https://open-eo.github.io/openeo-python-client/auth.html){target="_blank"}
+has a more in-depth information of various authentication concepts,
+and discusses alternative authentication methods.
+
+
+## Non-interactive And Machine-to-Machine Authentication with the openEO Python client library
```

Comment:
This too seems a bit longer title to me

Reply:
To make it easy for people to find this I wanted to include "non-interactive" and "machine-to-machine" in the title. It also focuses heavily on python use cases, so I wanted to make that clear

Reply:
made it bit shorter